bugwarrior/services/gmail.py

```diff
@@ -151,40 +161,67 @@ def get_threads(self):
             thread_service.get(userId='me', id=thread['id']).execute()
             for thread in result.get('threads', [])]
 
+    def annotations(self, issue):
+        sender = issue.extra['last_sender_name']
+        subj = issue.extra['subject']
+        issue_url = issue.get_processed_url(issue.extra['url'])
+        return self.build_annotations([(sender, subj)], issue_url)
+
     def issues(self):
         labels = self.get_labels()
         for thread in self.get_threads():
-            yield self.get_issue_for_record(thread, thread_extras(thread, labels))
+            issue = self.get_issue_for_record(thread, thread_extras(thread, labels))
+            extra = {
+                'annotations': self.annotations(issue),
```

Comment:
Any reason not to add the annotations in the ```thread_extras``` function?

Reply:
My reasoning was that `thread_extras` grabs "extra" information specific to gmail threads, while annotations are more like "extra" information for a Taskwarrior task. Also, since I pass information pulled from `thread_extras` as inputs to `self.build_annotations`, it seemed cleaner to keep them as separate steps.

Reply:
Sure, sounds reasonable.